File: api/monhun/mhgu.js
const dataUtils = require('../../tools/utils/dataUtils');
const routeUtils = require('../../tools/utils/routeUtils');
const config = require('../../config.json');

class MHGURouter {
  constructor() {
    this.monster_map = dataUtils.buildMap(
      './databases/mh_data/mhgu/monster_info.json'
    );
    this.weapon_map = dataUtils.buildMap(
      './databases/mh_data/mhgu/weapon_info.json'
    );
  }

  init() {
    const self = this;

    let routes = {
      monsters: function () {
        routeUtils.get('/api/mhgu/monsters', '', function (render, req, res) {
          res.json(self.monster_map.raw);
        });
      },
      weapons: function () {
        routeUtils.get('/api/mhgu/weapons', '', function (render, req, res) {
          res.json(self.weapon_map.raw);
        });
      }
    };

    for (const i in routes) {
      routes[i]();
    }
  }
}

module.exports = new MHGURouter();

File: python-api/api/routers/mhgu.py
from fastapi import APIRouter

from ..util import json_from_file

MHGU_DB_PREFIX = "mh_data/mhgu"


router = APIRouter()


@router.get("/monsters")
async def get_monsters():
    return json_from_file(f"{MHGU_DB_PREFIX}/monster_info.json")


@router.get("/weapons")
async def get_weapons():
    return json_from_file(f"{MHGU_DB_PREFIX}/weapon_info.json")
